pyarr/sonarr_api.py

```diff
@@ -1,470 +1,677 @@
+        res = self.request_put(path, data=data)
+        return res
+
+    ## HISTORY
+
+    # GET /history
+    def get_history(self, sort_key, page=1, page_size=10, sort_dir="desc", id_=None):
+        """Gets history (grabs/failures/completed)
+
+        Args:
+            [Required] sort_key (str) - series.title or date
+            [Optional] page (int) - Default: 1
+            [Optional] page_size (int) - Default: 10
+            [Optional] sort_dir (str) - asc or desc - Default: desc
+            [Optional] id_ (int) - Filters to a specific episode ID
+        Returns:
+            JSON Response (dict)
+        """
+        path = "/api/history"
+        params = {
+            "sortKey": sort_key,
+            "page": page,
+            "pageSize": page_size,
+            "sortDir": sort_dir,
+        }
+        if id_:
+            params["episodeId"] = id_
+        res = self.request_get(path, params=params)
+        return res
+
+    ## WANTED (MISSING)
+
+    # GET /wanted/missing
+    def get_wanted(self, sort_key, page=1, page_size=10, sort_dir="asc"):
+        """Gets Wanted / Missing episodes
+
+        Args:
+            [Required] sort_key (str) - series.title or airDateUtc
+            [Optional] page (int) - Default: 1
+            [Optional] page_size (int) - Default: 10
+            [Optional] sort_dir (str) - asc or desc - Default: asc
+        Returns:
+            JSON Response (dict)
+        """
+        path = "/api/wanted/missing"
+        params = {
+            "sortKey": sort_key,
+            "page": page,
+            "pageSize": page_size,
+            "sortDir": sort_dir,
+        }
+        res = self.request_get(path, params=params)
+        return res
+
+    ## QUEUE
+
+    # GET /queue
+    def get_queue(self):
+        """Gets current downloading info
+
+        Args:
+            None
+        Returns:
+            JSON Response (dict)
+        """
+        path = "/api/queue"
         res = self.request_get(path)
         return res
 
+    # DELETE /queue
+    def del_queue(self, id_, blacklist=True):
+        """Deletes an item from the queue and download client.
+        Optionally blacklist item after deletion.
+
+        Args:
+            [Required] id_ (int)
+            [Optional] blacklist (bool) - Default: True
+        Returns:
+            JSON Response (dict)
+        """
+        params = {"id": id_, "blacklist": blacklist}
+        path = "/api/queue/"
+        res = self.request_del(path, params=params)
+        return res
+
+    ## PARSE
+
+    # GET /parse
+    def get_parsed_title(self, title):
+        """Returns the result of parsing a title.
+
+        Args:
+            [Required] title (str)
+        Returns:
+            JSON Response (dict)
+        """
+        params = {"title": title}
+        path = "/api/parse"
+        res = self.request_get(path, params=params)
+        return res
+
+    # GET /parse
+    def get_parsed_path(self, path):
```

Comment:
same error as for the parsed_title

Reply:
Unknown why this one fails - I'll need to test this in the future.